src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4856,11 +4856,19 @@ uint os::processor_id() {
 }
 
 void os::set_native_thread_name(const char *name) {
+  char buf[16]; // according to glibc manpage, 16 chars incl. '/0'
+  // We may need to truncate the thread name. Since a common pattern is
+  // for thread names is to be both longer than 15 chars and have a trailing number
+  // ("DispatcherWorkerThread21", "C2 CompilerThread#54" etc), we truncate "smartly"
+  // by attempting to preserve the trailing number in the name if there is one
```

Comment:
This "smart" aspect is no longer present right? If the name ends in a number the number will be presented (or at least a few digits) simply because it comes at the end.

There will always be cases where the truncation method results in a "worse" name than an alternate truncation method would - hence why I do not think it is worth over engineering this. For example if the name is `ForkJoinPool-5-worker-21` you will lose important information no matter what.

Reply:
This series of ASAN issues came from a real customer case we are working on, and in that one having the trailing number preserved was really useful. And certainly, something like "C2 Com..hread4" is more useful than "C2 CompilerThre"?